ua/variant.go

```diff
@@ -347,6 +457,7 @@ func (m *Variant) Float() float64 {
 	}
 }
 
+// Int returns the int value if the type is one of the int types.
```

Comment:
minor nit on the comment/api: this returns an `int` _no matter what_. I'd rather see it return an error if it is not possible to "cast" it to the requested type. I know this makes the API even more verbose, but that's the pain of dynamic typing. It hasn't bit me yet, but I am 100% that in the future I will mistakenly ask for the wrong type and it won't be caught in any tests because it returns a real value.
I suppose we could go crazy and add `MustInt() int` and panic on error ;)

Reply:
While going through the code to unexport the variant fields I've noticed that in a lot of places we're just calling `Value()` instead of the specific method. I had the opposite reaction than you. 

I think that in almost all cases you know which type you're getting and the convenience functions should return that typed value without error checking. If you need to check you can always use `Value()` and to a proper type check. 

We should add methods for all supported types like `LocalizedText` and `XmlElement`, ... to complete the set. We could also add `Int8()` and the likes but that might be overkill.

Reply:
Noob OPC question: do (server) types change much? Meaning if I write a client assuming that the temperature is coming back as a float, and then an operator changes it to an int, my code will continue to work, but the temperature is now zero.

Reply:
I am running into the same issue with our PLC programmers. The answer is "it depends". There is this concept of a TypeDictionary but that seems quite cumbersome and we don't have code to use that. Schema versioning is a difficult problem and it usually requires coordination. 

Also, I don't want to add too much boilerplate code to the library if you can write a simple function yourself to handle this situation.

Reply:
Also, the same thing happens when they add a new field and don't tell you about it.